app/views/units/create.js.erb

```diff
@@ -0,0 +1 @@
+$("#contents").append("<li id=\"unit-<%= @unit.id %>\"> <%=@unit.name%> <a data-remote=\"true\" rel=\"nofollow\" data-method=\"delete\" href=\"/units/<%=@unit.id%>\">Delete Unit</a>");
```

Comment:
avoid `js.erb` interactions. Use JSON and AJAX, write JavaScript that responds to the resultant object that comes from the server.